### PR TITLE
add very basic metrics

### PR DESCRIPTION
**Pull request by rajatjindal**
Added very basic metrics to the aws-encryption-provider to get sense of how many encrypt/decrypt calls are successful/failed.

### CODE REVIEW

**Comment by sethpollack** [comment]
Thanks @rajatjindal. 

@rajatjindal  @micahhausler  Thoughts on moving metrics to its own package, initializing/registering in `cmd/server/main.go` rather then with init and then passing to `plugin.New`?

**Comment by rajatjindal** [comment]
thanks for quick review @sethpollack . I am open to learning new pattern for this. 

just to understand correctly, are you saying we will pass all metrics vars to the New function and store them in the *Plugin object?

do u have any sample example demonstrating this, i will go through that.

File: pkg/plugin/metrics.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+func registerPrometheusMetrics() {
+	prometheus.MustRegister(encryptionSuccessfulCount)
+	prometheus.MustRegister(encryptionFailedCount)
+	prometheus.MustRegister(decryptionSuccessfulCount)
+	prometheus.MustRegister(decryptionFailedCount)
+}
+
+var (
+	encryptionSuccessfulCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
```
[the remark this answers is not in the document]
Comment (rajatjindal) [reply pkg/plugin/metrics.go]: yes, that is way better. i'll push the change

**Comment by rajatjindal** [comment]
> Thanks @rajatjindal.
> 
> @rajatjindal @micahhausler Thoughts on moving metrics to its own package, initializing/registering in `cmd/server/main.go` rather than with init and then passing to `plugin.New`?

also @micahhausler can you please provide feedback on this.

**Comment by micahhausler** [comment]
> > Thanks @rajatjindal.
> > @rajatjindal @micahhausler Thoughts on moving metrics to its own package, initializing/registering in `cmd/server/main.go` rather than with init and then passing to `plugin.New`?
> 
> also @micahhausler can you please provide feedback on this.

I'm fine with it as-is, because at the end of the day we're using the global [`prometheus.defaultRegistry`](https://github.com/prometheus/client_golang/blob/HEAD//prometheus/registry.go#L54) which adds the `NewProcessCollector()` and `NewGoCollector()`. @sethpollack do you see a need to run multiple `Plugin`s at the same time and differentiate?

**Comment by sethpollack** [comment]
Ok, works for me.

File: pkg/plugin/metrics.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+func registerPrometheusMetrics() {
+	prometheus.MustRegister(kmsOperationCounter)
+}
+
+var (
+	kmsOperationCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "aws_encryption_provider_operations_total",
```
Comment (micahhausler) [inline pkg/plugin/metrics.go]: nit: could you rename this to `aws_encryption_provider_kms_operations_total`? It just clarifies that this metric is tracking `kms` metrics

Comment (rajatjindal) [reply pkg/plugin/metrics.go]: sure thing. done

**Comment by rajatjindal** [comment]
sample output:

```
aws_encryption_provider_operations_total{key_arn="arn:aws:kms:us-east-1:111112222333:key/222222-11111-11111-2222-11111111",operation="decrypt",status="success"} 12
aws_encryption_provider_operations_total{key_arn="arn:aws:kms:us-east-1:111112222333:key/222222-11111-11111-2222-11111111",operation="encrypt",status="success"} 38
```

**Pull request merged**